app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/KeyGeneratorAdapter.java

```diff
@@ -28,4 +36,41 @@ private KeyGeneratorAdapter(final IServiceLocatorCreator serviceLocatorCreator)
     public static ModuleWrapper<KeyGeneratorAdapter> newInstance(final IServiceLocatorCreator serviceLocatorCreator) {
         return new ModuleWrapper<>(new KeyGeneratorAdapter(serviceLocatorCreator));
     }
+
+    /**
+     * Generates a random KeyPair with the RSA algorithm.
+     * @return a randomly generated KeyPair
+     */
+    public static KeyPair generateRandomKeyPair() throws NoSuchAlgorithmException {
+        final KeyPairGenerator k = KeyPairGenerator.getInstance("RSA");
+        k.initialize(2048);
+
+        return k.generateKeyPair();
+    }
+
+    /**
+     * Gets from the parameter 'kp' the public key and makes a RSAPublicKeySpec out of it.
+     * @param keyPair the KeyPair with the public key
+     * @return the RSAPublicKeySpec
+     */
+    public static RSAPublicKeySpec getPublicKeySpec(final KeyPair keyPair) throws NoSuchAlgorithmException,
```

Comment:
Are we interested in the `RSAPublicKeySpec` (and to an extend the `RSAPrivateKeySpec`)? Don't we just want to call the `generateRandomKeyPair()` and get a usable public/private KeyPair (probably our own implementation where the keys are directly accessible as strings)?